### ISSUE TITLE
zig build segfault

### ISSUE THREAD

**Issue by marler8997** [ziglang/zig#3058]
I get a segfault when running `zig build` on the "zig" branch in my project: 

https://github.com/marler8997/audio

Reproduce on linux with:
```bash
git clone https://github.com/marler8997/audio
cd audio
git fetch origin zig:zig
git checkout zig
cd zig
zig build
```

It fails pretty quick with the release compiler, but with the debug compiler that I built it runs and takes about 2.5G of memory, then my computer grinds to a halt for a few minutes as starts thrashing memory (I have 4GB of ram). Eventually it finishes and gives this output:
```
Segmentation fault at address 0x0
/home/marler8997/git/zig/src/analyze.cpp:1504:25: 0xea38a3 in type_is_invalid (/home/marler8997/git/zig/src/analyze.cpp)
    switch (type_entry->id) {
                        ^
/home/marler8997/git/zig/src/analyze.cpp:5036:24: 0xeada66 in type_resolve (/home/marler8997/git/zig/src/analyze.cpp)
    if (type_is_invalid(ty))
                       ^
/home/marler8997/git/zig/src/analyze.cpp:7269:5: 0xeb4ca0 in get_llvm_type (/home/marler8997/git/zig/src/analyze.cpp)
    assertNoError(type_resolve(g, type, ResolveStatusLLVMFull));
    ^
/home/marler8997/git/zig/src/analyze.cpp:6690:46: 0xeb264d in resolve_llvm_types_union (/home/marler8997/git/zig/src/analyze.cpp)
        union_type->llvm_type = get_llvm_type(g, tag_type);
                                             ^
/home/marler8997/git/zig/src/analyze.cpp:7226:44: 0xeb4a43 in resolve_llvm_types (/home/marler8997/git/zig/src/analyze.cpp)
            return resolve_llvm_types_union(g, type, wanted_resolve_status);
                                           ^
/home/marler8997/git/zig/src/analyze.cpp:5072:31: 0xeadc1a in type_resolve (/home/marler8997/git/zig/src/analyze.cpp)
            resolve_llvm_types(g, ty, status);
                              ^
/home/marler8997/git/zig/src/analyze.cpp:7276:5: 0xeb4d84 in get_llvm_di_type (/home/marler8997/git/zig/src/analyze.cpp)
    assertNoError(type_resolve(g, type, ResolveStatusLLVMFull));
    ^
/home/marler8997/git/zig/src/codegen.cpp:578:53: 0xddd826 in get_di_scope (/home/marler8997/git/zig/src/codegen.cpp)
                scope->di_scope = ZigLLVMTypeToScope(get_llvm_di_type(g, decls_scope->container_type));
                                                    ^
/home/marler8997/git/zig/src/codegen.cpp:560:55: 0xddd6da in get_di_scope (/home/marler8997/git/zig/src/codegen.cpp)
            ZigLLVMDIScope *fn_di_scope = get_di_scope(g, scope->parent);
                                                      ^
/home/marler8997/git/zig/src/codegen.cpp:602:32: 0xddd941 in get_di_scope (/home/marler8997/git/zig/src/codegen.cpp)
            return get_di_scope(g, scope->parent);
                               ^
/home/marler8997/git/zig/src/codegen.cpp:587:72: 0xddd8da in get_di_scope (/home/marler8997/git/zig/src/codegen.cpp)
            ZigLLVMDILexicalBlock *di_block = ZigLLVMCreateLexicalBlock(g->dbuilder,
                                                                       ^
/home/marler8997/git/zig/src/codegen.cpp:5434:35: 0xdf0ae9 in set_debug_location (/home/marler8997/git/zig/src/codegen.cpp)
    ZigLLVMSetCurrentDebugLocation(g->builder, (int)source_node->line + 1,
                                  ^
/home/marler8997/git/zig/src/codegen.cpp:5717:35: 0xdf1638 in ir_render (/home/marler8997/git/zig/src/codegen.cpp)
                set_debug_location(g, instruction);
                                  ^
/home/marler8997/git/zig/src/codegen.cpp:6879:18: 0xdf6302 in do_code_gen (/home/marler8997/git/zig/src/codegen.cpp)
        ir_render(g, fn_table_entry);
                 ^
/home/marler8997/git/zig/src/codegen.cpp:9572:24: 0xdff4e1 in codegen_build_and_link (/home/marler8997/git/zig/src/codegen.cpp)
            do_code_gen(g);
                       ^
/home/marler8997/git/zig/src/main.cpp:1207:39: 0xdd52b6 in main (/home/marler8997/git/zig/src/main.cpp)
                codegen_build_and_link(g);
                                      ^
???:?:?: 0x7f8c2a4a9b8d in ??? (???)


???:?:?: 0x17c6258d4c544154 in ??? (???)


The following command terminated unexpectedly:
/home/marler8997/git/zig/build/bin/zig build-exe /home/marler8997/git/audio/zig/main.zig --cache-dir /home/marler8997/git/audio/zig/zig-cache --name audio --cache on 

Build failed. The following command failed:
/home/marler8997/git/audio/zig/zig-cache/o/-2O1d-4SPCwq29jpgVWshfbgyDYCl973twqNO0MT1Sx5BKIg-fxRmeZvEaevyX2D/build /home/marler8997/git/zig/build/bin/zig /home/marler8997/git/audio/zig /home/marler8997/git/audio/zig/zig-cache
```

**Comment by marler8997** [comment]
Test case reduced to this:
```zig
test "union no tag with struct member" {
    const Struct = struct { };
    const Union = union {
        s : Struct,
        pub fn foo(self: *@This()) void { }
    };
    var u = Union { .s = Struct {} };
    u.foo();
}
```
Compiling this code causes zig to segfault.

Fixed Here: https://github.com/ziglang/zig/pull/3119

**Comment by andrewrk** [comment]
fix landed in 42cc4a406bd4d037f4203fa2ebca2853db33c780
